src/java/io/bazel/rulesscala/scalac/ScalacProcessor.java

```diff
@@ -58,6 +58,9 @@ public void processRequest(List<String> args) throws Exception {
       List<File> javaJarFiles = filterFilesByExtension(jarFiles, ".java");
 
       String[] scalaSources = GenericWorker.appendToString(ops.files, scalaJarFiles);
+      // include java files from .srcjar's to support mixed compilation
+      scalaSources = GenericWorker.appendToString(scalaSources, javaJarFiles);
```

Comment:
I think a small private method which handles `srcjar`s (both scala and java) would be somewhat better since it can remove the need for the comment.
You can decide to have the new variable inside that method to also strengthen the point about mixed sources. I'll not insist but I think it will be clearer.